### ISSUE TITLE
Upgrade to Dinero.js 2.0.0

### ISSUE THREAD

**Issue by jeremyckahn** [jeremyckahn/farmhand#171]
Per [the library author's request](https://twitter.com/frontstuff_io/status/1422961817039552516), upgrade to Dinero.js 2.0.0 and share feedback with that project.

/cc @sarahdayan

**Comment by sarahdayan** [comment]
I'd love to help with that, but I'm also interested in your feedback on the migration process! This is helpful to improve the [migration guide](https://v2.dinerojs.com/docs/getting-started/upgrade-guide) based on real world experience.

I'd also recommend waiting a bit until v2.0.0-alpha.8 because I'm about to release breaking changes (see https://github.com/dinerojs/dinero.js/pull/309). I'd love to know what you think of this change, since you're using `toUnit` in your code.

**Comment by jeremyckahn** [comment]
Thanks for chiming jn @sarahdayan! I’ve subscribed to https://github.com/dinerojs/dinero.js/pull/309 and will pick this issue up once v2.0.0-alpha.8 is ready to go. I’ll let you know how the migration process goes for me!

**Comment by sarahdayan** [comment]
Cool! I will push another PR to allow for `transformScale` to take a rounding strategy since for now it won't be available anymore in `toUnits`. Both features will be in v2.0.0-alpha.8. Will tag you there.

**Comment by jeremyckahn** [comment]
@sarahdayan I just noticed https://github.com/dinerojs/dinero.js/releases/tag/v2.0.0-alpha.8! I'll try to get this working in Farmhand. 🙂

**Comment by sarahdayan** [comment]
@jeremyckahn Hey! I didn't yet ship the PR in this release unfortunately (needed to release a few bug fixes, and I want to test the PR some more). Y can definitely start and if you're done before v2.0.0-alpha.9 lands, I'll help with handling the breaking changes :)

**Comment by jeremyckahn** [comment]
@sarahdayan got it. I guess I jumped the gun a bit. 🙂 I started the migration to `v2.0.0-alpha.8` but ran into a few issues. It'll probably be easiest to wait for a more stable version (which it sounds like `v2.0.0-alpha.9` will hopefully be) and try migrating again then. Please let me know when you've got a version that you think is ready to be migrated to!

**Comment by sarahdayan** [comment]
@jeremyckahn What issues did you run into?

**Comment by jeremyckahn** [comment]
I mainly stumbled with getting used to 2.0's approach to handling internationalization and formatting. I also struggled a bit with changes to how rounding are handled. For instance, I noticed these tests failing during my (incomplete) migration efforts: https://github.com/jeremyckahn/farmhand/blob/ec62a3117bb5185934756a18ad0b4388c648ab08/src/utils.test.js#L96-L112

To be fair, the issues I had may not have been due to bugs with 2.0 or reflective of any design problems, but rather the limited time I've had to experiment with migration so far. 🙂 I just figured it would be a better use of time to stop and wait until you're feeling good about an alpha release for me to try out.

In case you're curious, I think the only parts of Farmhand's codebase that are directly concerned with Dinero.js are here: https://github.com/jeremyckahn/farmhand/blob/ec62a3117bb5185934756a18ad0b4388c648ab08/src/utils.js#L165-L194

and here: https://github.com/jeremyckahn/farmhand/blob/ec62a3117bb5185934756a18ad0b4388c648ab08/src/utils.js#L208-L222

Is there anything here that you think might be problematic to migrate to 2.0, @sarahdayan?

**Comment by jeremyckahn** [comment]
@sarahdayan this is great!! Thank you for taking the time to write up such a thorough and detailed guide for migrating Farmhand. What you've described and shared makes a lot of sense and should help make upgrading to Dinero.js 2.0 pretty straightforward. As soon as I have some time, I'll get a draft PR started with your suggested changes and wait for https://github.com/dinerojs/dinero.js/pull/309 to be merged. Once it is, I'll update my branch and promote it to a proper PR and tag you in it.

>I'm not sure exactly why you need to convert it to precision: 0 first though, as the mask returns the amount with no sub-units.

I'm not sure why I did that either. 😅 It looks like I added that `.convertPrecision(0)` a little over a year ago: https://github.com/jeremyckahn/farmhand/commit/803ce0dc637ee208ff86bf1295e1f4bcb0322352#diff-3274f1a37032fb0ae4e2823def0007c634e869ae0dfc304ff6a12c36513c3a52R81-R84. Removing it doesn't seem to break any tests, so it may have a been misunderstanding of the Dinero.js API, or just overly-defensive programming on my part. In any case, it's good to get assurance from the author that it's not needed. 🙂

>Let me know if you have questions or feedback on how to improve the docs and migration guide!

It's really well done so far! Thanks for putting all the work you have into it. My time for OSS work is pretty limited these days so I haven't been able to go through as carefully and methodically as I'd like, but the guide has definitely been useful. I hope Farmhand can serve as a helpful case study for your Dinero.js 2.0 migration efforts!

**Comment by lstebner** [comment]
is this stale? I see a PR was opened at one point, but closed.

**Comment by jeremyckahn** [comment]
Yeah I think we can close this out. Currency formatting is working as desired in Farmhand, albeit with outdated techniques. Specifically, we would be better served by using https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat, but that's not what this ticket is about.
